src/reanimated2/animation/transformationMatrix/matrixUtils.tsx

```diff
@@ -0,0 +1,369 @@
+
+  a.forEach((_, i) => {
+    switch (operation) {
+      case 'subtract':
+        a[i] -= b[i];
+        break;
+      case 'add':
+        a[i] += b[i];
+        break;
+    }
+  });
+
+  return isFlatOnStart ? (a as T) : (unflatten(a) as T);
+}
+
+export function subtractMatrices<T extends AffiniteMatrixFlat | AffiniteMatrix>(
+  _a: T,
+  _b: T
+): T {
+  'worklet';
+
+  return linearOperationOnMatrices(_a, _b, 'subtract');
+}
+
+export function addMatrices<T extends AffiniteMatrixFlat | AffiniteMatrix>(
+  _a: T,
+  _b: T
+): T {
+  'worklet';
+
+  return linearOperationOnMatrices(_a, _b, 'add');
+}
+
+export function scaleMatrix<T extends AffiniteMatrixFlat | AffiniteMatrix>(
+  _a: T,
+  scalar: number
+): T {
+  'worklet';
+
+  return linearOperationOnMatrix(_a, (x) => x * scalar);
+}
+
+export function getRotationMatrix(
+  angle: number,
+  axis: 'x' | 'y' | 'z' = 'z'
+): AffiniteMatrix {
+  'worklet';
+  const cos = Math.cos(angle);
+  const sin = Math.sin(angle);
+  switch (axis) {
+    case 'z':
+      return [
+        [cos, sin, 0, 0],
+        [-sin, cos, 0, 0],
+        [0, 0, 1, 0],
+        [0, 0, 0, 1],
+      ];
+    case 'y':
+      return [
+        [cos, 0, -sin, 0],
+        [0, 1, 0, 0],
+        [sin, 0, cos, 0],
+        [0, 0, 0, 1],
+      ];
+    case 'x':
+      return [
+        [1, 0, 0, 0],
+        [0, cos, sin, 0],
+        [0, -sin, cos, 0],
+        [0, 0, 0, 1],
+      ];
+  }
+}
+
+function norm(x: number, y: number, z: number) {
+  'worklet';
+  return Math.sqrt(x * x + y * y + z * z);
+}
+
+function transposeMatrix(_m: AffiniteMatrix): AffiniteMatrix {
+  'worklet';
+  const m = flatten(_m);
+  return [
+    [m[0], m[4], m[8], m[12]],
+    [m[1], m[5], m[9], m[13]],
+    [m[2], m[6], m[10], m[14]],
+    [m[3], m[7], m[11], m[15]],
+  ];
+}
+
+function innerProduct(a: number[], b: number[]) {
+  'worklet';
+  if (!(a.length === b.length)) {
+    console.warn(
+      `Error when evaluation inner product of ${a} and ${b}, because these arrays should have equal length`
+    );
+    return 0;
+  }
+  let output = 0;
+  a.forEach((_, i) => (output += a[i] * b[i]));
+  return output;
+}
+
+function projection(u: number[], a: number[]) {
+  'worklet';
+  const scaler = innerProduct(u, a) / innerProduct(u, u);
+  const output: Array<number> = [];
+  u.forEach((e) => output.push(e * scaler));
+  return output;
+}
+
+function subtract(a: number[], b: number[]) {
+  'worklet';
+  const output: Array<number> = [];
+  a.forEach((_, i) => output.push(a[i] - b[i]));
+  return output;
+}
+
+function scale(u: number[], a: number) {
+  'worklet';
+  const output: Array<number> = [];
+  u.forEach((e) => output.push(e * a));
+  return output;
+}
+
+function grahamSchmidtAlghoritm(matrix: AffiniteMatrix): {
+  rotationMatrix: AffiniteMatrix;
+  skewMatrix: AffiniteMatrix;
+} {
+  'worklet';
+  const a1 = matrix[0];
+  const a2 = matrix[1];
+  const a3 = matrix[2];
+  const a4 = matrix[3];
```

Comment:
```suggestion
  const [a1, a2, a3, a4] = matrix;
```
BTW maybe we should use either 0-3 or 1-4 but consistently?